src/tribol/interface/tribol.cpp

```diff
@@ -921,6 +1110,96 @@ integer update( integer cycle, real t, real &dt )
       }
 
       CouplingScheme* couplingScheme  = csManager.getCoupling(csIndex);
+
+      // update redecomp meshes if supplied mfem data
+      if (couplingScheme->hasMfemData())
+      {
+         auto mfem_data = couplingScheme->getMfemMeshData();
+         axom::Array<int> mesh_ids {2, 2};
+         mesh_ids[0] = mfem_data->GetMesh1ID();
+         mesh_ids[1] = mfem_data->GetMesh2ID();
+         mfem_data->UpdateMeshData();
+         auto coord_ptrs = mfem_data->GetRedecompCoordsPtrs();
+
+         // save penalty data for old meshes
```

Comment:
Clarify what is meant by 'old meshes'. If the penalty data has been changed between successive calls to tribol::update(), wouldn't the m_elemData you are grabbing have the latest and greatest data?

Reply:
Penalty data is no longer automatically updated.  The host code should set penalty parameters in each timestep (or cycle).